### PR TITLE
Csharp73 pattern based fixed statement

### PR DESCRIPTION
**Pull request by BillWagner**
Add a sample that uses the pattern based fixed statement with `Span<T>`

This makes use of the pattern `DangerousGetPinnableReference`

For dotnet/docs#4774

### CODE REVIEW

File: snippets/csharp/keywords/FixedKeywordExamples.cs [hunk 1]
```diff
@@ -39,6 +40,27 @@ unsafe private static void ModifyFixedStorage()
...
+            int[] PascalsTriangle = {
+                          1,
+                        1,  1,
+                      1,  2,  1,
+                    1,  3,  3,  1,
+                  1,  4,  6,  4,  1,
+                1,  5,  10, 10, 5,  1
+            };
+
+            Span<int> RowSix = new Span<int>(PascalsTriangle, 10, 5);
```
Comment (pkulikov) [inline snippets/csharp/keywords/FixedKeywordExamples.cs]: It looks to be `RowFive` that starts at element 10 and contains 5 elements.
Or let's update argument values:
```csharp
Span<int> RowSix = new Span<int>(PascalsTriangle, 15, 6);
```

Comment (BillWagner) [reply snippets/csharp/keywords/FixedKeywordExamples.cs]: Yup, I changed the variable name.

File: snippets/csharp/keywords/FixedKeywordExamples.cs [hunk 2]
```diff
@@ -39,6 +40,27 @@ unsafe private static void ModifyFixedStorage()
...
+                    1,  3,  3,  1,
+                  1,  4,  6,  4,  1,
+                1,  5,  10, 10, 5,  1
+            };
+
+            Span<int> RowFive = new Span<int>(PascalsTriangle, 10, 5);
+
+            fixed (int* ptrToRow = RowFive)
+            {
+                // Use the numbers 1, 5, 10, 10, 5, 1
```
Comment (pkulikov) [inline snippets/csharp/keywords/FixedKeywordExamples.cs]: Then here we can use numbers 1, 4, 6, 4, 1?

Comment (rpetrusha) [inline snippets/csharp/keywords/FixedKeywordExamples.cs]: The comment is wrong -- you're taking elements 10 through 14, which are 1,4,6,4,1.
Did you want to do anything with the numbers in the fixed block?

**Comment by rpetrusha** [comment]
You can ignore my comments, @BillWagner -- you addressed them in your commit at about the same time I was submitting my comments. LGTM now.

**Pull request merged**